### PR TITLE
Configure Travis CI with Clang builds

### PR DESCRIPTION
**Pull request by rakhimov**
Issue #11

### CODE REVIEW

**Comment by ogoffart** [comment]
The only combinaison that fails is the combinaison Qt 5.5 and Clang.
It fails because qmake passes  -std=c++11 instead of c++1y.

I think that Qt 5.5 detects at qt configure time if its compiler supports C++14,  and apparently, the if the compiler used to build qt did not support it c++1y, it would use c++11.  But then, why is it working with gcc?

**Comment by rakhimov** [comment]
quirks w/ clang + libstdc++?

**Comment by rakhimov** [comment]
Does this apply: https://stackoverflow.com/a/32552807

**Comment by ogoffart** [comment]
I don't think it's related to libstdc++/libc++

The problem is the file `mkspecs/features/c++14.prf` in Qt 5.5

```
} else: clang {
    # Clang has supported -std=c++1y since version 3.2
    greaterThan(QT_CLANG_MAJOR_VERSION, 3)|greaterThan(QT_CLANG_MINOR_VERSION, 1): \
```

However the QT_CLANG_MAJOR_VERSION / QT_CLANG_MINOR_VERSION are set by the Qt's configure script. That's why it works with gcc, because Qt probably was configured with GCC 4.8 already had the c++1y flag. But it was not configured with clang so that can't work.

So the options are:

1. We remove that configuration (Qt 5.5 and clang).  There is already enough tests in the matrix.
2. We add `QMAKE_CXXFLAGS=-std=c++1y` to the qmake line for this configuration.


In parallel, i have the same issue with Qt 5.2 which i'm trying to support: qmake from Qt 5.2 did not support the `CONFIG+=c++14` so we also would need to pass it manually: 
https://travis-ci.org/woboq/verdigris/jobs/326490115

**Comment by rakhimov** [comment]
qmake is driving me krazy: https://travis-ci.org/rakhimov/verdigris/jobs/326505668
The QMAKE_CXXFLAGS seem to have hit the same bug.

**Comment by rakhimov** [comment]
The range-v3 library also  requires extended constexpr:
https://github.com/ericniebler/range-v3/#supported-compilers

It recommends at-least clang 3.6 ,
so maybe dropping 3.5 is the way to go.

**Comment by rakhimov** [comment]
ok, removed Qt 5.5 and clang config; it's now passing.

File: .travis.yml [hunk 1]
```diff
@@ -12,26 +12,43 @@ env:
...
-    - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-${GCC_VERSION} 90;
-    - source /opt/qt$QT_VERSION/bin/qt${QT_VERSION}-env.sh;
-    - g++ --version
+    - source /opt/qt$QT_VERSION/bin/qt${QT_VERSION}-env.sh
 
 script:
     - which qmake
     - qmake --version
-    - qmake -r
+    - qmake -r "QMAKE_CXXFLAGS=-std=c++1y"
```
Comment (ogoffart) [inline .travis.yml]: Please remove this then, if it is not needed.

**Comment by rakhimov** [comment]
I added a single macos build that runs w/ clang & latest qt5 from homebrew.
Unfortunately, it is the last build in the matrix and queued most of the time.
We could make the macos build the first,
but I don't know a clean way of doing it (only through some ugly hackery).

**Pull request merged**